index.bs

```diff
@@ -795,6 +728,584 @@ the following list:
+that represents the information associated with closing a WebSocket.
+
+<xmp class="idl">
+[Exposed=(Window,Worker)]
+interface WebSocketError : DOMException {
+  constructor(optional DOMString message = "",
+              optional WebSocketCloseInfo init = {});
+
+  readonly attribute unsigned short? closeCode;
+  readonly attribute USVString reason;
+};
+</xmp>
+
+Each {{WebSocketError}} object has an associated <dfn for="WebSocketError">closeCode</dfn>, which is
+a number.
+
+Each {{WebSocketError}} object has an associated <dfn for="WebSocketError">reason</dfn>, which is a
+string.
+
+<dl class="domintro non-normative">
+ : <code>|error| = new {{WebSocketError/constructor(message, init)|WebSocketError}}([|message| [,
+   |init| ] ]</code>
+ :: Creates a new {{WebSocketError}} object.
+
+  |message| is a string which will be used to initialize the {{DOMException/message}} attribute of
+  the base class.
+
+  The |init| argument is an object whose properties can be set as follows:
+
+  : {{WebSocketCloseInfo/closeCode}}
+  :: A number, either 1000 or any integer in the range 3000 to 4999, inclusive. Any other number
+   will result in an "{{InvalidAccessError}}" {{DOMException}}.
+  : {{WebSocketCloseInfo/reason}}
+  :: A string. Must be 123 bytes or less when converted to [=UTF-8=]. A longer string will result in
+   a "{{SyntaxError}}" {{DOMException}} being thrown.
+
+ : <code>|error|.{{WebSocketError/closeCode}}</code>
+ :: Returns the [=the WebSocket connection close code=].
+
+ : <code>|error|.{{WebSocketError/reason}}</code>
+ :: Returns the [=the WebSocket connection close reason=].
+</dl>
+
+<hr>
+
+<div algorithm>
+
+The <dfn constructor for="WebSocketError" data-lt="WebSocketError(message, init)"><code>new
+WebSocketError(|message|, |init|)</code></dfn> constructor steps are:
+
+ 1. Set [=this=]'s [=DOMException/name=] to "<code>WebSocketError</code>".
+ 1. Set [=this=]'s [=DOMException/message=] to |message|.
+ 1. Let |code| be |init|["{{WebSocketCloseInfo/closeCode}}"] if it [=map/exists=], or unset
+    otherwise.
+ 1. Let |reason| be |init|["{{WebSocketCloseInfo/reason}}"] if it [=map/exists=], or unset
+    otherwise.
+ 1. [=Validate close code and reason=] with |code| and |reason|.
+ 1. If |reason| is non-empty, but |code| is not set, then set |code| to 1000 ("Normal Closure").
+ 1. Set [=this=]'s [=WebSocketError/closeCode=] to |code|.
+ 1. Set [=this=]'s [=WebSocketError/reason=] to |reason|.
+</div>
+
+The <dfn attribute for=WebSocketError>closeCode</dfn> getter steps are to return [=this=]'s
+[=WebSocketError/closeCode=].
+
+The <dfn attribute for=WebSocketError>reason</dfn> getter steps are to return [=this=]'s
+[=WebSocketError/reason=].
+
+# Common algorithms # {#common-algorithms}
+
+These algorithms are shared between the {{WebSocket}} and {{WebSocketStream}} interfaces.
+
+<div algorithm>
+
+To <dfn>get a URL record</dfn> given a |url| and |baseURL|,
```

Comment:
```suggestion
To <dfn>get a URL record</dfn> given a |url| and |baseURL|:
```

here and elsewhere. (See https://infra.spec.whatwg.org/#algorithm-declaration )

Reply:
Done, thanks.